paasta_tools/cli/cmds/logs.py

```diff
@@ -356,171 +293,268 @@ def prettify_log_line(line, requested_levels):
     return pretty_line
 
 
-def tail_paasta_logs(service, levels, components, clusters, raw_mode=False):
-    """Sergeant function for spawning off all the right log tailing functions.
+# The map of name -> LogReader subclasses, used by configure_log.
+_log_reader_classes = {}
 
-    NOTE: This function spawns concurrent processes and doesn't necessarily
-    worry about cleaning them up! That's because we expect to just exit the
-    main process when this function returns (as main() does). Someone calling
-    this function directly with something like "while True: tail_paasta_logs()"
-    may be very sad.
 
-    NOTE: We try pretty hard to supress KeyboardInterrupts to prevent big
-    useless stack traces, but it turns out to be non-trivial and we fail ~10%
-    of the time. We decided we could live with it and we're shipping this to
-    see how it fares in real world testing.
+def register_log_reader(name):
+    """Returns a decorator that registers that bounce function at a given name
+    so get_log_reader_classes can find it."""
+    def outer(bounce_func):
+        _log_reader_classes[name] = bounce_func
```

Comment:
This confuses me. Is this bounce-specific? Could this decorator not be applied to any function? (I don't understand what the bounce has to do with it)